pkg/systemd/init.js

```diff
@@ -369,15 +381,48 @@ $(function() {
             render();
         }
 
-        ReactDOM.render(
-            React.createElement(ServiceTabs, {
-                onChange: tab_changed,
-            }),
-            document.getElementById("services-filter"));
+        let tab_warnings = { };
+
+        function render_tabs() {
+            ReactDOM.render(
+                React.createElement(ServiceTabs, {
+                    onChange: tab_changed,
+                    warnings: tab_warnings,
+                }),
+                document.getElementById("services-filter"));
+        }
+
+        render_tabs();
 
         $("#services-text-filter").on("input", render);
         $(document).on("click", "#clear-all-filters", clear_filters);
 
+        function process_failed_units() {
+            const failed = new Set();
+            tab_warnings = { };
+            for (const p in units_by_path) {
+                const u = units_by_path[p];
+                if (u.ActiveState == "failed") {
+                    failed.add(u.Id);
+                    tab_warnings[u.Id.substr(u.Id.lastIndexOf('.') + 1)] = true;
```

Comment:
This could do with a comment how the Ids look like and what this does.

Also, feels a bit odd to mix Set() and objects with `true` values here (tab_warnings is also conceptually a set); but this is just bikeshedding, and good enough.

Reply:
Yeah, I see your point... but. :-)